Python/bytecodes.c

```diff
@@ -5150,21 +5146,15 @@ dummy_func(
                 int optimized = _PyOptimizer_Optimize(frame, target, &executor, 0);
                 if (optimized <= 0) {
                     exit->temperature = restart_backoff_counter(exit->temperature);
-                    if (optimized < 0) {
-                        GOTO_UNWIND();
-                    }
-                    GOTO_TIER_ONE(target);
-                }
-                else {
-                    exit->temperature = initial_temperature_backoff_counter();
+                    GOTO_TIER_ONE(optimized < 0 ? NULL : target);
                 }
+                exit->temperature = initial_temperature_backoff_counter();
```

Comment:
This looks correct, but it is a change in behavior. What impact to do expect this to have?

Reply:
It shouldn't be a change in behavior, right? `GOTO_TIER_ONE` breaks out of this instruction, so the `else` was always unnecessary.

Reply:
Yes, I misread the code.